core/src/main/java/com/linecorp/armeria/common/stream/DefaultStreamMessage.java

```diff
@@ -402,6 +426,41 @@ private boolean setState(State oldState, State newState) {
     }
 
     private void cleanup() {
-        cleanupQueue(subscription, queue);
+        final Throwable cause = ClosedStreamException.get();
```

Comment:
`cause` could be instantiated only when necessary.

Reply:
Ah fixed. 😄